lib/spack/spack/cmd/style.py

```diff
@@ -33,7 +33,8 @@ def grouper(iterable, n, fillvalue=None):
     "Collect data into fixed-length chunks or blocks"
     # grouper('ABCDEFG', 3, 'x') --> ABC DEF Gxx"
     args = [iter(iterable)] * n
-    return zip_longest(*args, fillvalue=fillvalue)
+    for group in zip_longest(*args, fillvalue=fillvalue):
+        yield filter(None, group)
```

Comment:
`filter(None, x)` is the same as `filter(lambda e: e is not None, x)`, which we do in every method that calls this iterator, so it seemed reasonable to move it up here.